editor/editor_file_dialog.cpp

```diff
@@ -624,7 +628,11 @@ void EditorFileDialog::_item_list_item_rmb_selected(int p_item, const Vector2 &p
 	}
 }
 
-void EditorFileDialog::_item_list_rmb_clicked(const Vector2 &p_pos) {
+void EditorFileDialog::_item_list_rmb_empty_clicked(const Vector2 &p_pos, MouseButton p_mouse_button_index) {
+	if (p_mouse_button_index != MouseButton::RIGHT) {
+		return;
+	}
```

Comment:
This shouldn't actually be here. The commend below is misleading, LMB click also allows for deselecting files.
The signal is emitted with any button, so not sure if other clicks than LMB or RMB should deselect too.

Reply:
<!-- Error while uploading GIF-220408_092023.gif -->
But this is working

Reply:
But it's working
![GIF-220408_092023](https://user-images.githubusercontent.com/79760854/162359639-7a1bb28a-6a38-448f-9c69-2b06f2be92b9.gif)

Reply:
This is Tree. Check the file dialog when you do e.g. Save As.

Reply:
Strange, should I change it to `p_mouse_button_index == MouseButton::MIDDLE` and method name to `_item_list_empty_clicked`

Reply:
It should be `(p_mouse_button_index != MouseButton::RIGHT && p_mouse_button_index != MouseButton::LEFT)`. Also make sure mouse left doesn't open context menu.

Reply:
Nope, it was not the problem. Actually I forgot to take care of `nothing_selected`(emits on lmb click) which is now merged into `empty_clicked`. I have fixed it but there's a problem, there's is `_update_selection_pattern_from_tileset_pattern_selection` in tile_map_editor.cpp which is called for `empty_clicked`,  `item_activated` and `item_selected`, this method doesn't has any arguments, I didn't know we can call a function in C++ without passing all the arguements or it's something else? as `item_activated` and `item_selected` do have one arguments. For `empty_clicked` we must check for mouse button being left as previously it was `nothing_selected`. 
I don't know how to deal with it, could you please instruct me?

Reply:
You will need to add intermediate method, e.g. `patterns_item_list_empty_clicked()` that calls `_update_selection_pattern_from_tileset_pattern_selection()` if clicked with lmb.

Reply:
> You will need to add intermediate method, e.g. `patterns_item_list_empty_clicked()` that calls `_update_selection_pattern_from_tileset_pattern_selection()` if clicked with lmb.
> 

Oops that was a silly qu


> You will need to add intermediate method, e.g. `patterns_item_list_empty_clicked()` that calls `_update_selection_pattern_from_tileset_pattern_selection()` if clicked with lmb.
> 

Oops, that was a silly question.

Reply:
> You will need to add intermediate method, e.g. `patterns_item_list_empty_clicked()` that calls `_update_selection_pattern_from_tileset_pattern_selection()` if clicked with lmb.
> 

Oops, that was a silly question.

Reply:
> btw I checked your recent file manager changes (is it final?) and left click opens context menu.

My bad, fixed that. Now the PR is final.